src/assisted_test_infra/test_infra/helper_classes/base_cluster.py

```diff
@@ -162,21 +161,12 @@ def set_installer_args(self):
     def to_cluster_hosts(hosts: list[dict[str, Any]]) -> list[ClusterHost]:
         return [ClusterHost(models.Host(**h)) for h in hosts]
 
-    def find_matching_node_name(self, host: ClusterHost, nodes: list[Node]) -> Union[str, None]:
+    def find_matching_node(self, host: ClusterHost, nodes: list[Node]) -> Optional[Node]:
         # Looking for node matches the given host by its mac address (which is unique)
         for node in nodes:
             for mac in node.macs:
                 if mac.lower() in host.macs():
-                    return node.name
-
-        # IPv6 static ips
-        if self._infra_env_config.is_static_ip:
-            mappings = static_network.get_name_to_mac_addresses_mapping(self.nodes.controller.tf_folder)
```

Comment:
this part of the code doesn't work: `self.nodes.controller.tf_folder`, `nodes` is a `List`, not a `Node`.
I deleted it as we don't test the ipv6 anymore.